client/src/pages/contentPages/Groups.js

```diff
@@ -1,41 +1,326 @@
+    ...(isDragAccept ? acceptStyle : {}),
+    ...(isDragReject ? rejectStyle : {})
+  }), [
+    isDragActive,
+    isDragReject,
+    isDragAccept
+  ]);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setGroupPicture('');
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setGroupPicture('');
+    setGroupName('');
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    
+    if(formValid) {
+
+      const data = { groupName: groupName, imageUrl: groupPicture , courseId: courseId};
+      const response = await axios.post('/user/groups', data);
+
+      if(response.data) {
+        enqueueSnackbar('Group Created Successfully.', {
+          variant: 'success',
+          autoHideDuration: '5000',
+        });
+        setOpen(false)
+      } else {
+        enqueueSnackbar('Error Creating Group.', {
+          variant: 'Error',
+          autoHideDuration: '5000',
+        });
+      }
+      
+    } else {
+      enqueueSnackbar('Error Creating Group.', {
+          variant: 'Error',
+          autoHideDuration: '5000',
+      });
+    }
+
+  };
+
+  const handleOnChange = async (e) => {
+    checkGroupExists(e.target.value.toLowerCase())
+    
+  }
+
+const checkGroupExists = async (name) => {
+  try {
+
+    setGroupName(name)
+
+    if(groupNames.includes(name) !== true) {
+      setGroupErrors('');
+      setFormValid(true);
+      return false;
+    } else {
+      setGroupErrors("A group with this name already exists!");
+      setFormValid(false);
+      return true;
+    }
+
+  } catch (err) {
+      console.log(err);
+  }
+};
+
+const getUserGroups = async () => {
+  const res = await axios.get('/user/groups')
+  const groupNames = []
+  res.data.forEach(course => {
+    course.groups.forEach(group => {
+      groupNames.push(group.name.toLowerCase())
+    })
+ 
+  })
+  setGroupNames(groupNames)
+}
+
+  useEffect(() => {
+    setCourses(userCourse.userCourses);
+    setProfile(profile);
+    getUserGroups()
+    setUploading(false)
+  }, []);
+
+    const body = (
+      <div style={modalStyle} className={classes.paper}>
+      <h2>Create a new group</h2>
+      <p>
+        Add group name and select your course.
+      </p>
+      <form className={classes.form} onSubmit={handleSubmit}>
+      <FormHelperText>{groupError ? <span style={{color: "#fc2525"}}>{groupError}</span> : 'Please enter a group name'}</FormHelperText>
+      <TextField
+        name="group_name"
+        fullWidth={true}
+        variant="outlined"
+        onChange={handleOnChange}
+      >
+      </TextField>
+
+      <Box className={classes.groupImageContainer}>
+        <FormHelperText>Drag and Drop Group Picture </FormHelperText>
+          <Tooltip
+            title='Drag and drop profile picture'
+            arrow placement='right'
+          >
+          <Box {...getRootProps({style})}>
+            <img
+              alt='Profile Pic'
+              src={groupPicture.length ? groupPicture : defaultImage}
```

Comment:
Not a big deal because you've given image a default value on the backend, but a tip I got from @amha19, this can be changed to just `groupPicture ?  groupPicture: defaultImage` because an empty string still evaluates to False. And it's a bit safer because it wont create an error if groupPicture is undefined for some reason.